### ISSUE TITLE
[Bug]: CTC and w2v2 with fp16 is going NaN 

### ISSUE THREAD

**Issue by TParcollet** [speechbrain/speechbrain#1803]
### Describe the bug

When fine-tuning an ASR model based on CTC only and a w2v2 from HF (for instance CommonVoice/ASR/CTC), the loss will explode to NaN in few batches when fp16 is activated.

I suspect that this is due to the CTC loss and log_softmax being done in fp16 while they should be computed in fp32.

Don't have time to fix this now. 

### Expected behaviour

Training without NaN

**Comment by Adel-Moumen** [comment]
I was able to reproduce the bug. 

I started digging into the problem and found that GradScaler was the source of NaNs. 

I will get back to you when I find the reason/solution to fix the problem. :-)

**Comment by JuanPZuluaga** [comment]
@TParcollet did you solve this issue? Or simply moving out the loss computation from autocast (as now in `core.py` see below) solved the issue? 

https://github.com/speechbrain/speechbrain/blob/891318f5950c337bb951912bf64bd5973af7c908/speechbrain/core.py#L937

**Comment by Adel-Moumen** [comment]
Closing this issue as #2010 is merged in develop.
